tools/logger/convert.c

```diff
@@ -284,6 +285,18 @@ static void process_params(struct proc_ldc_entry *pe,
 			p += 2;
 		}
 	}
+
+	/* ensure no NULLs in format strings */
+	for (j = 0; j < i; j++) {
+		if (!pe->params[j]) {
+			pe->params[j] = (uintptr_t)log_asprintf("<missing>");
+			pe->subst_mask |= 1 << j;
```

Comment:
you could just use a single `static const char missing[] = "<missing>";` string and not set `subst_mask`

Reply:
@lyakh Actually yes, that would be potentially cleaner. This still requires mucking with the subst_mask (which is a bit iffy as in theory subst_mask is not just to indicate how the object was allocated). Let me update the patch.